apps/member/src/components/group/ActivityPostEditor/ActivityPostEditor.tsx

```diff
@@ -16,23 +31,68 @@ interface ActivityPostEditorProps {
 }
 
 const ActivityPostEditor = ({ groupId, data }: ActivityPostEditorProps) => {
+  const toast = useToast();
+  const { openModal } = useModal();
   const [post, setPost] = useState({
-    title: '제목을 입력해주세요.',
-    content: '내용을 입력해주세요.',
+    title: '',
+    content: '',
   });
+  const [editAssignment, setEditAssignment] = useState<boolean[]>(
+    new Array(data.length).fill(false),
+  );
+
+  const { activityGroupBoardMutate } = useActivityGroupBoardMutation();
+  const { activityGroupBoardDeleteMutate } =
+    useActivityGroupBoardDeleteMutation();
+  const { data: assignmentData } = useActivityGroupBoardByCategory(
+    groupId,
+    'ASSIGNMENT',
+  );
 
   const handlePostChange = (
     e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
   ) => {
     const { name, value } = e.target;
     setPost((prev) => ({ ...prev, [name]: value }));
   };
+  const handleAddWeeklyClick = () => {
+    if (!post.title || !post.content)
+      return toast({
+        state: 'error',
+        message: '제목, 내용은 필수 입력 요소입니다.',
+      });
+
+    const activityBoardItem: SubmitBoardType = {
+      category: 'WEEKLY_ACTIVITY',
+      ...post,
+    };
+    activityGroupBoardMutate({
+      activityGroupId: groupId,
+      body: activityBoardItem,
+    });
+  };
+  const handleDeleteWeeklyClick = (activityGroupBoardId: number) => {
+    activityGroupBoardDeleteMutate(activityGroupBoardId);
+  };
+  const handleEditWeeklyClick = (prevData: ActivityBoardType) => {
+    return openModal({
+      title: '수정하기',
+      custom: <ActivityBoardEditModal groupId={groupId} prevData={prevData} />,
+    });
+  };
```

Comment:
단순 return 구문만 존재하는거라면 아래와 같이 작성하는게 더 깔끔할 것 같네요

```suggestion
  const handleEditWeeklyClick = (prevData: ActivityBoardType) => openModal({
      title: '수정하기',
      custom: <ActivityBoardEditModal groupId={groupId} prevData={prevData} />,
    });
 
```